packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -832,38 +670,86 @@ export const validateSections = <TSection extends FieldSection>(
   }
 };
 
-export const mergeDateIntoReferenceDate = <
-  TDate,
-  TSection extends Omit<FieldSection, 'start' | 'end' | 'startInInput' | 'endInInput'>,
->(
+const transferDateSectionValue = <TDate>(
```

Comment:
I tried to make this function simpler to understand now that it is the only one doing transformation directly on a date.
No more lookup with complex typing, I just have one dumb `switch`